scss/_reboot.scss

```diff
@@ -373,147 +381,159 @@ select {
-button::-moz-focus-inner,
-[type="button"]::-moz-focus-inner,
-[type="reset"]::-moz-focus-inner,
-[type="submit"]::-moz-focus-inner {
+
+::-moz-focus-inner {
   padding: 0;
   border-style: none;
 }
 
-input[type="radio"],
-input[type="checkbox"] {
-  box-sizing: border-box; // 1. Add the correct box sizing in IE 10-
-  padding: 0; // 2. Remove the padding in IE 10-
-}
 
+// Remove the default appearance of temporal inputs to avoid a Mobile Safari
+// bug where setting a custom line-height prevents text from being vertically
+// centered within the input.
+// See https://bugs.webkit.org/show_bug.cgi?id=139848
+// and https://github.com/twbs/bootstrap/issues/11266
 
 input[type="date"],
 input[type="time"],
 input[type="datetime-local"],
 input[type="month"] {
-  // Remove the default appearance of temporal inputs to avoid a Mobile Safari
-  // bug where setting a custom line-height prevents text from being vertically
-  // centered within the input.
-  // See https://bugs.webkit.org/show_bug.cgi?id=139848
-  // and https://github.com/twbs/bootstrap/issues/11266
   -webkit-appearance: listbox;
 }
 
+// 1. Remove the default vertical scrollbar in IE.
+// 2. Textareas should really only resize vertically so they don't break their (horizontal) containers.
+
 textarea {
-  overflow: auto; // Remove the default vertical scrollbar in IE.
-  // Textareas should really only resize vertically so they don't break their (horizontal) containers.
-  resize: vertical;
+  overflow: auto; // 1
+  resize: vertical;  // 2
 }
 
+// 1. Browsers set a default `min-width: min-content;` on fieldsets,
+//    unlike e.g. `<div>`s, which have `min-width: 0;` by default.
+//    So we reset that to ensure fieldsets behave more like a standard block element.
+//    See https://github.com/twbs/bootstrap/issues/12359
+//    and https://html.spec.whatwg.org/multipage/#the-fieldset-and-legend-elements
+// 2. Reset the default outline behavior of fieldsets so they don't affect page layout.
+
 fieldset {
-  // Browsers set a default `min-width: min-content;` on fieldsets,
-  // unlike e.g. `<div>`s, which have `min-width: 0;` by default.
-  // So we reset that to ensure fieldsets behave more like a standard block element.
-  // See https://github.com/twbs/bootstrap/issues/12359
-  // and https://html.spec.whatwg.org/multipage/#the-fieldset-and-legend-elements
-  min-width: 0;
-  // Reset the default outline behavior of fieldsets so they don't affect page layout.
-  padding: 0;
-  margin: 0;
-  border: 0;
+  min-width: 0; // 1
+  padding: 0;  // 2
+  margin: 0; // 2
+  border: 0; // 2
 }
 
-// 1. Correct the text wrapping in Edge and IE.
-// 2. Correct the color inheritance from `fieldset` elements in IE.
+
+// 1. Correct the color inheritance from `fieldset` elements in IE.
+// 2. Correct the text wrapping in Edge and IE.
+
 legend {
   display: block;
   width: 100%;
-  max-width: 100%; // 1
```

Comment:
I found a issue: 
https://jsbin.com/wobotesige/edit?html,css,output

I think we should set `display: table` and `max-width: 100%;`.
Please see https://github.com/necolas/normalize.css/blob/8.0.1/normalize.css#L243-L244

Reply:
I see, `display: table` looks a bit better indeed.

Or we might just move the legend inside the fieldset? See https://jsbin.com/caqijiqiyo/1/edit?html,css,output

Reply:
If we support the aim of ​​Normalize.css:

> Preserve useful browser defaults rather than erasing them.
http://nicolasgallagher.com/about-normalize-css/

that position should be left as it is.

Reply:
Nope, we don't, reboot adds some opinionated styling (https://getbootstrap.com/docs/4.3/content/reboot/): 
> Reboot builds upon Normalize, providing many HTML elements with somewhat opinionated styles using only element selectors

I think moving the legend inside the fieldset (last example https://jsbin.com/caqijiqiyo/1/edit?html,css,output) will simplify development for custom theming.

But I don't really have a strong opinion on this. @mdo, what do you think?

Reply:
@MartijnCuppens Yup, this aims to be much more opinionated than Normalize. I could be down with the `float` method here to further simplify things, but I'm wondering if that doesn't just sign us up for more resetting when folks do other things with `<legend>`s?

Reply:
Since Bootstrap removes the borders by default and tries to undo the typical behaviour of legends, I've added the `float: left`. Now the fieldsets and legend both behave like block elements and can easily be themed with utilities if needed.